app/GioView.java

```diff
@@ -554,8 +554,11 @@ private class GioInputConnection implements InputConnection {
 
 		@Override public int getCursorCapsMode(int reqModes) {
 			Snippet snip = getSnippet();
-			int selStart = imeSelectionStart(nhandle);
-			return TextUtils.getCapsMode(snip.snippet, imeToUTF16(nhandle, selStart), reqModes);
+			int selStart = imeToUTF16(nhandle, imeSelectionStart(nhandle));
+			if (selStart > snip.snippet.length()) {
+			    selStart = 0;
```

Comment:
What if `snip.snippet` is empty? I suggest just `return 0` here, because the caps mode is undefined at positions outside the snippet.

Reply:
Thinking a bit more, we control `selStart,` so why is it ever greater than `snip.snippet.length`? Something must be updating either the selection or the snippet without adjusting the other.

Reply:
Thinking a bit more, we control selStart, so why is it ever greater than snip.snippet.length? Something must be updating either the selection or the snippet without adjusting the other.

Reply:
Fixed in tip.